lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1712,7 +1721,7 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
 
         onehotTy = rewriter.getType<Torch::ValueTensorType>(
             onehotShape, resultType.getDtype());
-        onehot = rewriter.create<Torch::AtenWhereScalarOp>(loc, onehotTy,
+        onehot = rewriter.create<Torch::AtenWhereScalarOp>(loc, resultType,
```

Comment:
I'm not sure why the `onehotTy` was created, so can't comment on removing it either. It's better if you confirm this with @rsuderman before merging.

Reply:
This is addressed in PR #3333